src/eventSchemas/providers/schemasDataProvider.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Schemas } from 'aws-sdk'
+import _ = require('lodash')
+import { SchemaClient } from '../../shared/clients/schemaClient'
+import { toArrayAsync } from '../../shared/utilities/collectionUtils'
+
+export interface regionRegistryMap {
+    region: string
+    registrySchameMapList: registrySchemaMap[]
+}
+
+export interface registrySchemaMap {
+    registryName: string
+    schemaList: Schemas.SchemaSummary[]
+}
+
+/**
+ * Responsible for retaining registry && schema list per region for Create-New-SAM-Application wizard
+ */
+export class SchemasDataProvider {
+    private static INSTANCE: SchemasDataProvider | undefined
+    private readonly cachedRegions: regionRegistryMap[] = []
+
+    public async getRegistires(region: string, client: SchemaClient) {
+        const cachedRegion = this.cachedRegions.filter(cacheValue => cacheValue.region === region).shift()
+        let registryNames: string[] = []
+
+        // if region is cached, retrieve registries
+        if (cachedRegion) {
+            registryNames = cachedRegion.registrySchameMapList.reduce(
+                (accumulator: string[], item: registrySchemaMap) => {
+                    accumulator.push(item.registryName)
+
+                    return accumulator
+                },
+                []
+            )
```

Comment:
Can we cache the registryNames in a way that we don't have to iterate on them when retrieving from the cache?

Reply:
Yes definitely, will do that in my first commit.